core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -340,7 +365,7 @@ private[spark] object TaskMetrics extends Logging {
         externalAccums.add(tmpAcc)
       }
     }
-    tm._externalAccums.addAll(externalAccums)
+    tm._externalAccums ++= externalAccums.asScala
```

Comment:
I'm wondering if we still need the local `externalAccums`. `tm._externalAccums` is `ArrayBuffer` now and the one-by-one insertion is no longer an issue.

Reply:
Make sense. Will remove it.